portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -14,78 +14,87 @@
+  private List<Comment> getComments() {
+    // Loads comments from Datastore
+    List<Comment> comments = CommentDataStore.COMMENT_OBJECT_DATA_STORE.load();
+
+    return comments;
+  }
+
+
+  /** Converts the list of Comment objecct into json form */
   private String convertToJsonUsingGson(List<Comment> comments) {
     Gson gson = new Gson();
     String json = gson.toJson(comments);
     return json;
   }
 
+  /** Sends the JSON as response */
+  private void sendJsonResponse(HttpServletResponse response, String json) throws IOException {
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get comments from the form
-    String commenter = getParameter(request, /*name=*/"commenter", /*defaultValue=*/"");
-    String content = getParameter(request, /*name=*/"content", /*defaultValue=*/"No comments");
-    long timestamp = System.currentTimeMillis();
+    // Gets comments from the form
+    String commenter = ServletUtils.getParameter(
+            request, /*name=*/Constants.PARAM_NAME_COMMENTER, /*defaultValue=*/"");
+    String content = ServletUtils.getParameter(
+            request, /*name=*/Constants.PARAM_NAME_CONTENT, /*defaultValue=*/"No comments");
     // TODO: validate request parameters
 
-    // Modify the state of server.
-    addComment(commenter, content);
+    // Stores the comment into the Datastore
+    storeComment(commenter, content);
 
-    Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("commenter", commenter);
-    commentEntity.setProperty("content", content);
-    commentEntity.setProperty("timestamp", timestamp);
-
-    DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
-    datastoreService.put(commentEntity);
-
-    // Redirect back to the HTML page.
+    // Redirects back to the HTML page.
     response.sendRedirect("/index.html");
   }
 
-  private void addComment(String name, String comment) {
-    comments.add(new Comment(name, comment, LocalDateTime.now()));
-  }
+  /** Stores the comment into the Datastore */
+  private void storeComment(String commenter, String content) {
+    // Creates a miscellaneous comment object to convert it to entity
+    long timestamp = System.currentTimeMillis();
+    Comment comment = new Comment(commenter, content, timestamp);
 
-  /**
-   * @return the value of parameter with the {@code name} in the {@code request}
-   *         or returns {@code defaultValue} if that parameter does not exist.
-   */
-  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
-    String value = request.getParameter(name);
-    if (value == null) {
-      return defaultValue;
-    }
-    return value;
+    // Stores the comment as an entity into Datastore
+    CommentDataStore.COMMENT_OBJECT_DATA_STORE.store(comment);
```

Comment:
Actually during servlet init, maybe we can have this.commentDatastore = new CommentDatastore(underlyingDatastore); ? so in this method we just use it as datastore.store(comment)?

Reply:
Fixed